compat/mingw.c

```diff
@@ -2705,6 +2705,16 @@ int mingw_rename(const char *pold, const char *pnew)
 		if (CopyFileW(wpold, wpnew, FALSE) && !mingw_unlink(pold))
 			return 0;
 		gle = GetLastError();
+	} else if ((attrsold = GetFileAttributesW(wpold)) & FILE_ATTRIBUTE_READONLY) {
```

Comment:
Could we guard this behind the appropriate `gle` condition?

Reply:
It was `ERROR_ACCESS_DENIED`, so I've moved both the container exception and this one into a nested if. Is that OK?

Reply:
Yes!

I also fixed some white-space, squashed the commits into a single one, and added code to re-set the read-only flag also if the rename failed.

Reply:
Yeah, sorry about that. I was having a nightmare with my editor and whitespace. When I tried to switch to tabs, it marked the entire file as changed, no idea why...

And reset of course—I was staring at the code knowing there was something else to be handled, and just couldn't see it. Thanks for that. Although, I _suspect_ on any other filesystem, if the file is giving access denied, it would also prevent you from changing the attributes. Better safe though.